text/0001-script-based-tokenizer.md

```diff
@@ -0,0 +1,288 @@
+                if to_highlight { value.push_str("</mark>") }
+            } else {
+                value.push_str(token);
+            }
+        }
+    }
+}
+```
+
+#### Store
+
+```rust
+let tokenizer = Tokenizer::new(&content);
+for (pos, token_group) in tokenizer.filter_map(only_token).enumerate().take(MAX_POSITION) {
+    let position = (attr as usize * MAX_POSITION + pos) as u32;
+    for token in token_group.normalized() {
+        words_positions.entry(token.txt()).or_insert_with(SmallVec32::new).push(position);
+    }
+}
+```
+
+```rust
+// original
+for (pos, token) in simple_tokenizer(&content).filter_map(only_token).enumerate().take(MAX_POSITION) {
+    let word = token.to_lowercase();
+    let position = (attr as usize * MAX_POSITION + pos) as u32;
+    words_positions.entry(word).or_insert_with(SmallVec32::new).push(position);
+}
+```
+
+### Impact on documentation
+
+This feature should not impact the meilisearch user documentation,
+in future versions we will probably provide a way to configure tokenizer and this will be discussed in a new specification.
+
+## Technical Specifications
+
+### Architecture
+
+As the actual tokenizer, this tokenizer will be a standolone library,
+and will be used by meilisearch like another one.
+We decided to put it in an other repository to be able to use it in meilisearch/Meilisearch and in the future core-engine.
+
+### Implementation Details
+
+We could have 2 approach of tokenization based on the script,
+the main difficulty is to handle multi-script documents
+which need to segment each part of script and tokenize them differently,
+theses 2 approach are:
+- first iterate over the whole document, splitting it by script, and after, tokenize each part with specialized lexer
+- detect the script of the start of the document, tokenize with specialized lexer while script is the same, switch lexer if the script change
+
+#### Tokenizer
+
+```rust
+use crate::token::Token;
+use crate::internal_tokenizer::InternalTokenizer;
+
+struct Tokenizer<'a> {
+    /// script specialized tokenizer, this can be switched during
+    /// document tokenization if the document contains several scripts
+    current_tokenizer: Option<Box<dyn InternalTokenizer<'a>>>,
+    /// current character index in the document
+    current_char_index: u64,
+    /// reference on the document content
+    inner: &'a str,
+}
+
+impl<'a> Tokenizer<'a> {
+    /// create a new tokenizer detecting script
+    /// and chose the specialized internal tokenizer
+    fn new(inner: &'a str) -> Self { unimplemented!() }
+}
+
+impl<'a> Iterator for Tokenizer<'a> {
+    type Item = Token<'a>;
+
+    /// return the next token calling the `next` method of the internal tokenizer,
+    /// if the internal tokeizer return None, the function try to redetect script and chose a new tokenizer,
+    /// if no iternal tokenizer is chosen, the method return None
+    fn next(&mut self) -> Option<Self::Item> { unimplemented!() }
```

Comment:
Are you talking about switching the tokenizer inside the same attribute? It seems too much. I think we can put the limit at one script per field.